compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1818,9 +1819,17 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
    */
   def tryInsertApplyOrImplicit(tree: Tree, pt: ProtoType)(fallBack: => Tree)(implicit ctx: Context): Tree = {
 
+    def isSyntheticApply(tree: Tree): Boolean = tree match {
+      case tree: Select => tree.getAttachment(InsertedApply).isDefined
```

Comment:
Good question. It depends how we want to specity it: 

1st choice: An `apply` is inserted on any expression in function position unless that expression is already an inserted apply.

2nd choice: An `apply` is inserted on any expression in function position unless that expression is itself an `apply` selection or call.

Which way should we go? Here's where it makes a difference. Consider the case where we want to expand to

    f.apply(...).apply(...)

If we give only one `apply`, which one should be the inserted one? According to the PR, it must be the second `apply`. If we followed 2nd choice, this would be rejected instead.